api/v1alpha1/nnf_lustre_mgt.go

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2024 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	dwsv1alpha2 "github.com/DataWorkflowServices/dws/api/v1alpha2"
+	"github.com/DataWorkflowServices/dws/utils/updater"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NnfLustreMGTSpec defines the desired state of NnfLustreMGT
+type NnfLustreMGTSpec struct {
+	// Addresses is the list of LNet addresses for the MGT
+	Addresses []string `json:"addresses"`
+
+	// FsNameBlackList is a list of fsnames that can't be used. This may be
+	// necessary if the MGT hosts file systems external to Rabbit
+	FsNameBlackList []string `json:"fsNameBlackList,omitempty"`
+
+	// FsNameStart is the starting fsname to be used
```

Comment:
Use +kubebuilder:validation:MaxLength
https://book.kubebuilder.io/reference/markers/crd-validation

And fit it into other related places.